packages/web-features-figuring/src/baseline/index.ts

```diff
@@ -0,0 +1,244 @@
+import assert from "node:assert";
+
+import { Temporal } from "@js-temporal/polyfill";
+
+import { feature } from "../browser-compat-data/feature";
+import { Release } from "../browser-compat-data/release";
+import { browsers, highReleases, lowReleases } from "./core-browser-set";
+import { support } from "./support";
+import { Browser } from "../browser-compat-data/browser";
+import {
+  BASELINE_LOW_TO_HIGH_DURATION,
+  VERY_FAR_FUTURE_DATE,
+} from "../constants";
+import { Compat, defaultCompat } from "../browser-compat-data/compat";
+
+interface FeatureSelector {
+  compatKeys: [string, ...string[]];
+  checkAncestors: boolean;
+}
+
+interface SupportStatus {
+  compatKey?: string;
+  baseline: "low" | "high" | false;
+  baseline_low_date: string | null;
+  baseline_high_date: string | null;
+  support: Map<Browser, Release | undefined>;
+  toJSON: () => string;
+}
+
+export function computeBaseline(
+  featureSelector: FeatureSelector,
+  compat: Compat = defaultCompat,
+): SupportStatus {
+  const { compatKeys } = featureSelector;
+  const keys = featureSelector.checkAncestors
+    ? compatKeys.flatMap((key) => withAncestors(key, compat))
+    : compatKeys;
+
+  const statuses = keys.map((key) => calculate(key, compat));
+
+  const baseline = minStatus(statuses);
+
+  let baseline_low_date = null;
+  if (baseline !== false) {
+    baseline_low_date = latestDate(
+      statuses.map((status) => status.baseline_low_date),
+    );
+  }
+
+  let baseline_high_date = null;
+  if (baseline === "high") {
+    baseline_high_date = latestDate(
+      statuses.map((status) => status.baseline_high_date),
+    );
+  }
+
+  return {
+    baseline,
+    baseline_low_date,
+    baseline_high_date,
+    support: minSupport(statuses.map((status) => status.support)),
+    toJSON: function () {
+      return jsonify(this);
+    },
+  };
+}
+
+function calculate(compatKey: string, compat: Compat): SupportStatus {
+  const f = feature(compatKey);
+  const s = support(f, browsers(compat), compat);
+
+  const allReleases = f.supportedBy({ only: browsers(compat), compat });
```

Comment:
IIUC, this returns all of the releases that support the feature. Then we check if the latest browser releases (`lowReleases`) are a subset of this `allReleases`. Some comments to explain the rules would be helpful, since there are quite a few equivalent ways of crunching this data.